WalletWasabi/WabiSabi/Client/AmountDecomposer.cs

```diff
@@ -65,12 +65,49 @@ public IEnumerable<Money> Decompose(IEnumerable<Coin> myInputCoins, IEnumerable<
 				}
 			}
 
-			if (remaining >= MinimumAmountPlusFee)
+			if (remaining >= MinimumAmountPlusFee && remainingVsize >= OutputSize)
 			{
-				outputAmounts.Add(remaining - OutputFee);
+				naiveSet.Add(remaining - OutputFee);
+			}
+
+			setCandidates.Add(naiveSet.Count, naiveSet);
+			var before = DateTimeOffset.UtcNow;
+			do
+			{
+				remaining = inputs.Sum();
+				remainingVsize = AvailableVsize;
+				var currSet = new List<Money>();
+				do
+				{
+					var denomPlusFees = denoms.Where(x => x <= remaining && x >= (remaining / 3)).ToList();
+					var denomPlusFee = denomPlusFees.RandomElement();
+					if (denomPlusFee is null || remaining < MinimumAmountPlusFee || remainingVsize < 2 * OutputSize)
+					{
+						break;
```

Comment:
Same question for the last condition. Isn't this dangerous?

Reply:
It is OK to have it here, it just breaks the while. However I am not sure why 2 times? - safety margin? :)

Reply:
This isn't the dangerous parts like the other comments I had, the code just looks similar.